src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -189,7 +188,7 @@ private async Task ProcessQueueAsync()
                         var ranToCompletion = await work.CallbackAsync(context, cancellationToken).ConfigureAwait(false);
 
                         // Now that we've mutated our solution, clear out our saved state to ensure it gets recalculated
-                        _lastLSPSolution.Remove(context.Solution.Workspace);
+                        _lspSolutionCache.Remove(context.Solution.Workspace);
```

Comment:
any threading concerns?  or is this only going to execute one thing at a time?

Reply:
Yeah, this loop is the only thing that starts a new handler executing.